presto-hive/src/main/java/com/facebook/presto/hive/security/SqlStandardAccessControl.java

```diff
@@ -309,6 +335,10 @@ private boolean getGrantOptionForPrivilege(ConnectorTransactionHandle transactio
 
     private boolean checkTablePermission(ConnectorTransactionHandle transaction, Identity identity, SchemaTableName tableName, HivePrivilege... requiredPrivileges)
     {
+        if (tableName.equals(ROLES) && !isAdmin(transaction, identity)) {
```

Comment:
Throwing an exception is an antipattern for the information schema queries -- it would be better to filter out the roles table in the filterTables() method if the user isn't admin.